clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/userdata/LocalFileUserDataProvider.groovy

```diff
@@ -24,38 +26,80 @@ class LocalFileUserDataProvider implements UserDataProvider {
   @Autowired
   LocalFileUserDataProperties localFileUserDataProperties
 
+  @Autowired
+  Front50Service front50Service
+
+  boolean isLegacyUdf(String account, String applicationName) {
+    Closure<Boolean> result = {
+      List<Map> creds = front50Service.getCredentials()
+      Map cred = creds.find { it.global } ?: creds.find { it.name == account }
+      if (!cred) {
+        throw new IllegalStateException("Unable to find front50 credentials (global of $account)")
+      }
+      String front50Account = cred.name
+
+      try {
+        Map application = front50Service.getApplication(front50Account, applicationName)
+        if (application.legacyUdf == null) {
+          return localFileUserDataProperties.defaultLegacyUdf
+        }
+        return Boolean.valueOf(application.legacyUdf)
+      } catch (RetrofitError re) {
+        if (re.kind == RetrofitError.Kind.HTTP && re.response.status == 404) {
+          return localFileUserDataProperties.defaultLegacyUdf
+        }
+        throw re
+      }
+    }
+
+    final int maxRetry = 5
+    final int retryBackoff = 500
+    final Set<Integer> retryStatus = [429, 500]
+    for (int i = 0; i < maxRetry; i++) {
```

Comment:
probably an opportunity to address retry a bit more generally (I know there is also a bunch of custom retry logic in `DiscoverySupport`)